File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/BookManagement.py
class BookManagement:
    def __init__(self):
        self.inventory = {}

    def add_book(self, title: str, quantity: int):
        if title in self.inventory:
            self.inventory[title] += quantity
        else:
            self.inventory[title] = quantity

    def remove_book(self, title: str, quantity: int):
        if title not in self.inventory or self.inventory[title] < quantity:
            raise Exception("Invalid operation")
        new_quantity = self.inventory[title] - quantity
        if new_quantity == 0:
            del self.inventory[title]
        else:
            self.inventory[title] = new_quantity

    def view_inventory(self) -> dict:
        return dict(self.inventory)

    def view_book_quantity(self, title: str) -> int:
        return self.inventory.get(title, 0)

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_cpp/code_output/BookManagement.cpp
#include <unordered_map>
#include <string>
#include <stdexcept>

class BookManagement {
private:
    std::unordered_map<std::string, int> inventory;

public:
    BookManagement() = default;

    void addBook(const std::string& title, int quantity) {
        // operator[] inserts the key with a value-initialized int (0) if it doesn't exist,
        // which perfectly matches the behavior of Java's containsKey/put logic.
        inventory[title] += quantity;
    }

    void removeBook(const std::string& title, int quantity) {
        auto it = inventory.find(title);
        if (it == inventory.end() || it->second < quantity) {
            throw std::runtime_error("Invalid operation");
        }
        
        int newQuantity = it->second - quantity;
        if (newQuantity == 0) {
            inventory.erase(it);
        } else {
            it->second = newQuantity;
        }
    }

    std::unordered_map<std::string, int> viewInventory() const {
        // Returning by value creates a copy, identical to `new HashMap<>(inventory)` in Java
        return inventory;
    }

    int viewBookQuantity(const std::string& title) const {
        auto it = inventory.find(title);
        if (it != inventory.end()) {
            return it->second;
        }
        return 0;
    }
};